controller/src/main/java/io/pravega/controller/store/stream/records/RetentionStreamCutRecord.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream.records;
+
+import com.google.common.collect.ImmutableMap;
+import io.pravega.common.ObjectBuilder;
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import lombok.Builder;
+import lombok.Data;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * This is data class for storing stream cut with time when the cut was computed.
+ * And the size of data being cut.
+ * This record is indexed in retentionSet using the recording time and recording size.
+ */
+@Data
+@Builder
+@Slf4j
+public class RetentionStreamCutRecord {
+    public static final RetentionStreamCutRecordSerializer SERIALIZER = new RetentionStreamCutRecordSerializer();
+
+    /**
+     * Time when this stream cut was recorded.
+     */
+    final long recordingTime;
+    /**
+     * Amount of data in the stream preceeding this cut.
+     */
+    final long recordingSize;
+    /**
+     * Actual Stream cut.
+     */
+    final Map<StreamSegmentRecord, Long> streamCut;
+
+    public RetentionStreamCutRecord(long recordingTime, long recordingSize, Map<StreamSegmentRecord, Long> streamCut) {
+        this.recordingTime = recordingTime;
+        this.recordingSize = recordingSize;
+        this.streamCut = ImmutableMap.copyOf(streamCut);
```

Comment:
Yes, because otherwise the supplier of the map could change the map and it would get modified. 

What we want is that the data object should be immutable. So it gets a map as input and creates an immutable copy of it so that it is insulated from any changes to the original map.